sklearn/cluster/tests/test_spectral.py

```diff
@@ -191,6 +197,27 @@ def histogram(x, y, **kwargs):
         sp.fit(X)
 
 
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+def test_cluster_qr(dtype):
+    # Test cluster_qr for fixed data different dtypes return the same lables
+    random_state = np.random.RandomState(seed=8)
+    data = random_state.randn(10, 5).astype(dtype)
+    labels = cluster_qr(data)
+    assert np.array_equal(labels, np.array([2, 1, 3, 3, 2, 4, 1, 3, 4, 0]))
```

Comment:
@jjerphan The code you propose cannot run correctly since subsequent calls of random_state.randn will result in different output, even though the seed it fixed. I have not paid attention to it prior to committing, but it failed the test as it should have.

Anyway, it is being replaced with @ogrisel's test now